packages/sign_in_with_apple/android/src/main/kotlin/com/aboutyou/dart_packages/sign_in_with_apple/SignInWithApplePlugin.kt

```diff
@@ -53,62 +56,91 @@ public class SignInWithApplePlugin: FlutterPlugin, MethodCallHandler, ActivityAw
     @JvmStatic
     fun registerWith(registrar: Registrar) {
       val channel = MethodChannel(registrar.messenger(), "com.aboutyou.dart_packages.sign_in_with_apple")
-      channel.setMethodCallHandler(SignInWithApplePlugin())
+      val plugin = SignInWithApplePlugin()
+      channel.setMethodCallHandler(plugin)
+
+      registrar.addActivityResultListener(plugin)
+      registrar.activity()?.let(plugin::registerActivity)
+      registerLifecycle(application(registrar), plugin)
+    }
+
+    private fun registerLifecycle(app: Application, plugin: SignInWithApplePlugin) {
+      app.registerActivityLifecycleCallbacks(object : Application.ActivityLifecycleCallbacks {
+        override fun onActivityPaused(activity: Activity) = Unit
+        override fun onActivityResumed(activity: Activity) = Unit
+        override fun onActivityDestroyed(activity: Activity?) = plugin.unregisterActivity()
+        override fun onActivitySaveInstanceState(activity: Activity?, outState: Bundle?) = Unit
+        override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) = plugin.registerActivity(activity)
+        override fun onActivityStarted(activity: Activity) = Unit
+        override fun onActivityStopped(activity: Activity) = Unit
+      })
     }
+
+    private fun application(registrar: Registrar) = registrar.context().applicationContext as Application
+  }
+
+  fun registerActivity(currentActivity: Activity) {
+    activity = currentActivity
```

Comment:
Is there anything against directly setting the activity on the plugin instead of this function?

Reply:
Have done it this way for a better readability. But will revert.

Reply:
I did not want to make the `application` public. Is it OK with you or should I expose it?